java/flinktestmock/src/main/java/org/apache/flink/core/fs/FileStatus.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * This file is based on source code from the Hadoop Project (http://hadoop.apache.org/), licensed
+ * by the Apache Software Foundation (ASF) under the Apache License, Version 2.0. See the NOTICE
+ * file distributed with this work for additional information regarding copyright ownership.
+ */
+
+package org.apache.flink.core.fs;
+
+/**
+ * Interface that represents the client side information for a file independent of the file system.
+ */
+public interface FileStatus {
```

Comment:
I am thinking maybe we could just let ForSt provide all these classes as a complete jar in the future.
Then some duplicate codes could be avoided.

Reply:
Yes, I think we could provide flink-core and forst-statebackend jar packages for ForSt in the future.  At present, considering the problem of circular dependency, it needs to be bypassed in the current way.